drivers/gles2/rasterizer_canvas_gles2.h

```diff
@@ -129,7 +129,12 @@ class RasterizerCanvasGLES2 : public RasterizerCanvasBaseGLES2 {
 
 	// batch item may represent 1 or more items
 	struct BItemJoined {
-		uint32_t first_item_ref;
+		union {
+			uint32_t first_item_ref;
+			Item *single_item;
+		};
+		// >= 1: standard case, first_item_ref is valid
+		// 0: joining disabled, single_item is valid (this BItemJoined is used as an adapter to a single item)
 		uint32_t num_item_refs;
```

Comment:
Note that Item * will be 64 bit on 64 bit platforms, which will make the structure larger and affect alignment.